src/components/Header/Header.tsx

```diff
@@ -0,0 +1,134 @@
+import {
+  useCallback,
+  useEffect,
+  useState,
+} from 'react';
+import {
+  useLocation,
+  useParams,
+} from 'react-router-dom';
+import classNames from 'classnames';
+import { Logo } from '../Logo';
+import { MenuButton } from '../MenuButton';
+import { PageNav } from '../PageNav';
+import { TopActionButton } from '../TopActionButton';
+import { SearchBar } from '../SearchBar';
+import { useWindowSize } from '../../utils/useWindowSize';
+import productCategory from '../../api/productCategory.json';
+import './Header.scss';
+
+export const Header: React.FC = () => {
+  const location = useLocation();
+  const { productId = '' } = useParams();
+
+  const { width } = useWindowSize();
+
+  const [isOpened, setIsOpened] = useState(false);
+  const [isSearchBar, setIsSearchBar] = useState(false);
+
+  const productCategoryList = productCategory.map(item => item.type);
+
+  const searchBarCondition = productCategoryList.some(
+    item => location.pathname.includes(item),
+  ) && !productId;
+
+  const cartCondition = location.pathname.includes('cart');
+  const favoriteCondition = location.pathname.includes('favorite');
+
+  const handleIsOpened = useCallback((menuStatus: boolean) => {
```

Comment:
you will always toggle it to the opposit value, so I guess it might be called `toggleMenu`, etc., and toggle it here (shouldn't accept an argument)